mlir/lib/Conversion/GPUCommon/GPUToLLVMConversion.cpp

```diff
@@ -811,8 +811,13 @@ LogicalResult ConvertAllocOpToGpuRuntimeCallPattern::matchAndRewrite(
   // descriptor.
   Type elementPtrType = this->getElementPtrType(memRefType);
   auto stream = adaptor.getAsyncDependencies().front();
+
+  auto isHostShared = rewriter.create<mlir::LLVM::ConstantOp>(
+      loc, llvmInt64Type, rewriter.getI64IntegerAttr(isShared));
+
   Value allocatedPtr =
-      allocCallBuilder.create(loc, rewriter, {sizeBytes, stream}).getResult();
+      allocCallBuilder.create(loc, rewriter, {sizeBytes, stream, isHostShared})
```

Comment:
I thought we had a consensus about avoiding the use of `host_shared` and `async` because of USM's limitation in handling asynchronous data allocation. Are we still on the same page with this? 

https://github.com/llvm/llvm-project/pull/65539#discussion_r1320546363

Reply:
we need to relax the checks in that case in GPUToLLVMConversion Pass to allow lowering of non async gpu.alloc......also we might need to change the gpu-async-region pass to handle this.

Reply:
Sure, I think it's worth to do it! What's your take on this? Personally, I think having a complete PR is the way to go. Otherwise, we will have an improperly implemented Op.

Reply:
Thanks. any other feedback on the PR? or else we can merge it if it looks good to you